commit 0cf0c416ec73f65d4631ec8e2b6409f95462c76e
Author: msweet <msweet@a1ca3aef-8c08-0410-bb20-df032aa958be>
Date:   Thu Feb 20 16:09:55 2014 +0000

    Fix a D-BUS threading issue that caused the scheduler to crash (STR #4347)
    
    Apparently D-BUS is not thread-safe by default...
    
    
    git-svn-id: svn+ssh://src.apple.com/svn/cups/cups.org/trunk@11626 a1ca3aef-8c08-0410-bb20-df032aa958be

--- a/config-scripts/cups-common.m4
+++ b/config-scripts/cups-common.m4
@@ -303,16 +303,11 @@
 LIBS="$SAVELIBS"
 
 dnl Check for DBUS support
-if test -d /etc/dbus-1; then
-	DBUSDIR="/etc/dbus-1"
-else
-	DBUSDIR=""
-fi
-
 AC_ARG_ENABLE(dbus, [  --disable-dbus           build without DBUS support])
 AC_ARG_WITH(dbusdir, [  --with-dbusdir          set DBUS configuration directory ],
 	DBUSDIR="$withval")
 
+DBUSDIR=""
 DBUS_NOTIFIER=""
 DBUS_NOTIFIERLIBS=""
 
@@ -329,7 +324,12 @@
 		LIBS="$LIBS $DBUS_NOTIFIERLIBS"
 		AC_CHECK_FUNC(dbus_message_iter_init_append,
 			      AC_DEFINE(HAVE_DBUS_MESSAGE_ITER_INIT_APPEND))
+		AC_CHECK_FUNC(dbus_threads_init,
+			      AC_DEFINE(HAVE_DBUS_THREADS_INIT))
 		LIBS="$SAVELIBS"
+		if test -d /etc/dbus-1; then
+			DBUSDIR="/etc/dbus-1"
+		fi
 	else
 		AC_MSG_RESULT(no)
 	fi
--- a/config.h.in
+++ b/config.h.in
@@ -535,6 +535,7 @@
 
 #undef HAVE_DBUS
 #undef HAVE_DBUS_MESSAGE_ITER_INIT_APPEND
+#undef HAVE_DBUS_THREADS_INIT
 
 
 /*
--- a/scheduler/main.c
+++ b/scheduler/main.c
@@ -506,6 +506,14 @@
   setlocale(LC_TIME, "");
 #endif /* LC_TIME */
 
+#ifdef HAVE_DBUS_THREADS_INIT
+ /*
+  * Enable threading support for D-BUS...
+  */
+
+  dbus_threads_init();
+#endif /* HAVE_DBUS_THREADS_INIT */
+
  /*
   * Set the maximum number of files...
   */
